pkg/controller/atlas/provider.go

```diff
@@ -30,10 +32,16 @@ const (
 type Provider interface {
 	Client(ctx context.Context, secretRef *client.ObjectKey, log *zap.SugaredLogger) (*mongodbatlas.Client, string, error)
 	SdkClient(ctx context.Context, secretRef *client.ObjectKey, log *zap.SugaredLogger) (*admin.APIClient, string, error)
+	SdkClientSet(ctx context.Context, secretRef *client.ObjectKey, log *zap.SugaredLogger) (*ClientSet, string, error)
 	IsCloudGov() bool
 	IsResourceSupported(resource api.AtlasCustomResource) bool
 }
 
+type ClientSet struct {
+	//SdkClient20231115008 *admin.APIClient
+	SdkClient20241113001 *adminv20241113001.APIClient
+}
```

Comment:
So the plan is to keep removing clients from the Provider, such as `Client` and `SdkClient` and keep on managing the new ones only on the `ClientSet`?

I guess we do that to avoid having to change the client usages on exiting code.

Reply:
Yes, I think having a clientSet with a group of all clients could be a good solution, instead of extending the `Provider` interface every time we add a new version of Go SDK